packages/react/src/data-table/DataTable.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  type ColumnDef,
+  type SortingState,
+  flexRender,
+  getCoreRowModel,
+  getPaginationRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { assignInlineVars } from "@vanilla-extract/dynamic";
+import { createElement, useState } from "react";
+
+import { Box } from "../box";
+import { DataTableHeader } from "../data-table-header";
+import { Pagination } from "../pagination";
+import { Table } from "../table";
+import { TableBody } from "../table-body";
+import { TableCell } from "../table-cell";
+import { TableHead } from "../table-head";
+import { TableHeaderCell } from "../table-header-cell";
+import { TableRow } from "../table-row";
+import * as styles from "./DataTable.css";
+
+interface DataTableProps<TData, TValue> {
+  columns: ColumnDef<TData, TValue>[];
+  data: TData[];
+  pinnedColumns?: string[];
+}
+
+export type { ColumnDef };
```

Comment:
we don't need to re-export this from here - consumers should grab them directly from `@tanstack/react-table`

Reply:
Do we need to install `@tanstack/react-table` for storybook then?